tests/link.ts

```diff
@@ -0,0 +1,862 @@
+      const link = returnOne.concat(mock);
+
+      testLinkResults({
+        link,
+        results: [1],
+        done,
+      });
+    });
+
+    it('should concat a Link and function', (done) => {
+      const returnOne = new SetContextLink({ add: 1 });
+      const mock = new MockLink((op, forward) => {
+        let _op = {
+          ...op,
+          context: {
+            add: op.context.add + 2,
+          },
+        };
+        return forward(_op);
+      });
+      const link = returnOne.concat(mock).concat((op) => {
+        return Observable.of({ data: op.context.add });
+      });
+
+      testLinkResults({
+        link,
+        results: [3],
+        done,
+      });
+    });
+
+    it('should concat a function and Link', (done) => {
+      const returnOne = new SetContextLink({ add: 1 });
+      const mock = new MockLink((op, forward) => Observable.of({ data: op.context.add }));
+
+      const link = returnOne.concat((operation, forward) => {
+        operation = {
+          ...operation,
+          context: {
+            add: operation.context.add + 2,
+          },
+        };
+        return forward(operation);
+      }).concat(mock);
+      testLinkResults({
+        link,
+        results: [3],
+        done,
+      });
+    });
+
+    it('should concat two functions', (done) => {
+      const returnOne = new SetContextLink({ add: 1 });
+      const link = returnOne.concat((operation, forward) => {
+        operation = {
+          ...operation,
+          context: {
+            add: operation.context.add + 2,
+          },
+        };
+        return forward(operation);
+      }).concat((op, forward) => Observable.of({ data: op.context.add }));
+      testLinkResults({
+        link,
+        results: [3],
+        done,
+      });
+    });
+
+    it('should concat two Links', (done) => {
+      const returnOne = new SetContextLink({ add: 1 });
+      const mock1 = new MockLink((operation, forward) => {
+        operation = {
+          ...operation,
+          context: {
+            add: operation.context.add + 2,
+          },
+        };
+        return forward(operation);
+      });
+      const mock2 = new MockLink((op, forward) => Observable.of({ data: op.context.add }));
+
+      const link = returnOne.concat(mock1).concat(mock2);
+      testLinkResults({
+        link,
+        results: [3],
+        done,
+      });
+    });
+
+    it('should return an link that can be concat\'d multiple times', (done) => {
+      const returnOne = new SetContextLink({ add: 1 });
+      const mock1 = new MockLink((operation, forward) => {
+        operation = {
+          ...operation,
+          context: {
+            add: operation.context.add + 2,
+          },
+        };
+        return forward(operation);
+      });
+      const mock2 = new MockLink((op, forward) => Observable.of({ data: op.context.add + 2 }));
+      const mock3 = new MockLink((op, forward) => Observable.of({ data: op.context.add + 3 }));
```

Comment:
lets move `MockLink` to the `/tests` folder so it won't be confused as an external link